examples/react-native-project/.app-config.schema.yml

```diff
@@ -0,0 +1,10 @@
+type: object
+additionalProperties: false
```

Comment:
totally unrelated, but I've always pondered a `$object` directive to improve the ergonomics of schemas

Reply:
Do you have an example of what you mean?

Reply:
Without much design, something like

```
$object:
  foo: string
  bar: integer
```

Which would expand to the obvious JSON schema representation (as required fields). This is a can of worms (recreating a whole syntax), sort of like schema-builder.

Reply:
Gotcha. TBH this does feel out of scope & agreed would open a can of worms. Schema files are nice because they are (for the most part) standard JSON schema.

If boilerplate was a concern, I would sooner build a separate tool/standard that transpiles into JSON schema.

Or even better, perhaps we create a plug-in system around schema validation (JSON schema being the default, but you can replace with eg. Yup). We could also offer creating schema through eg. `.app-config.schema.js` at which you can generate using a schema builder and export.

Reply:
Yeah, I agree with all of those points.